### DISCUSSION TITLE
Bard: allow_source and floating toolbar not working together

### DISCUSSION THREAD

**Discussion by carstenjaksch** [statamic/cms#11226] [Q&A] [answered]
Hey,

I have noticed that toolbar mode "floating" and enabled "allow source" do not play well together.

The floating toolbar does not show the button to toggle between source and WYSIWYG view.

Should I raise an issue about that? Looks like a smallish bug to me.

**Comment by duncanmcclean** [comment, marked as answer]
It looks like it might have been [intentionally hidden](https://github.com/statamic/cms/commit/27df03b3df6433027f960123ed346dd5246f096b) at some point, likely to reduce the number of buttons being shown in smaller spaces.

To be honest, I don't think the "Source" option in Bard is _that_ useful. We're going to remove it.

**Reply by carstenjaksch** [reply]
Thank you for looking that up! So it is considered obsolete? I think the source view is quite useful to clean up markup. Coming from WordPress, TinyMCE easily messes things up. Is Bard different? 😄

**Reply by duncanmcclean** [reply]
Unless you have the "Save as HTML" option enabled, Bard doesn't save the content as HTML, unlike TinyMCE or editors in WordPress.

Instead, Bard saves the content in a "structured" manner, which makes it much easier to do things like sets/blocks where you're saving fields, not HTML. It also means that your content is more portable (eg. your content isn't locked to the current design, with inline CSS, etc).

If you *do* want to see the HTML output of a Bard field, you could open up Live Preview and see it.

**Reply by carstenjaksch** [reply]
Thank you for the explanation, I had a closer look and see now how the field is saved. Much better system, love it!
